src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java

```diff
@@ -137,6 +157,16 @@ void initializeReference() {
      */
     void initializeReads() {
         if (! readArguments.getReadFiles().isEmpty()) {
+            // Prefetcher is useful for cloud files because of the latencies involved.
+            Function<SeekableByteChannel, SeekableByteChannel> wrapper = Function.identity();
+            Function<SeekableByteChannel, SeekableByteChannel> indexWrapper = Function.identity();
+
+            if (cloudPrefetchBuffer > 0) {
+                wrapper = is -> GATKTool.addPrefetcher(cloudPrefetchBuffer, is);
+            }
+            if (cloudIndexPrefetchBuffer > 0) {
+                indexWrapper = is -> GATKTool.addPrefetcher(cloudIndexPrefetchBuffer, is);
+            }
```

Comment:
Let's move all this code to construct the prefetchers (including `addPrefetcher()` above and this code snippet here) down into `ReadsDataSource`, and just pass the buffer sizes as ints into the `ReadsDataSource` constructor.